image/nitf.ksy

```diff
@@ -0,0 +1,759 @@
+        type: length_reserved_info
+        repeat: expr
+        repeat-expr: num_reserved_extension.to_i
+      - id: user_defined_header
+        type: tre_header
+      - id: extended_header
+        type: tre_header
+  date_time:
+    seq:
+      - type: str
+        size: 14
+  encrypt:
+    seq:
+      - type: str
+        size: 1
+  clasnfo:
+    instances:
+      total_size: 
+        value: (1 + 2 + 11 + 2 + 20 + 2 + 8 + 4 + 1 + 8 + 43 + 1 + 40 + 1 + 8 + 15)
+    seq:
+      - id: sclas
+        type: str
+        size: 1
+      - id: sclsy
+        type: str
+        size: 2
+      - id: scode
+        type: str
+        size: 11
+      - id: sctlh
+        type: str
+        size: 2
+      - id: srel
+        type: str
+        size: 20
+      - id: sdctp
+        type: str
+        size: 2
+      - id: sdcdt
+        type: str
+        size: 8
+      - id: sdcxm
+        type: str
+        size: 4
+      - id: sdg
+        type: str
+        size: 1
+      - id: sdgdt
+        type: str
+        size: 8
+      - id: scltx
+        type: str
+        size: 43
+      - id: scatp
+        type: str
+        size: 1
+      - id: scaut
+        type: str
+        size: 40
+      - id: scrsn
+        type: str
+        size: 1
+      - id: ssrdt
+        type: str
+        size: 8
+      - id: sctln
+        type: str
+        size: 15
+  length_image_info:
+    seq:
+      - id: lish
+        type: str
+        size: 6
+      - id: li
+        type: str
+        size: 10
+  length_graphic_info:
+    seq:
+      - id: lssh
+        type: str
+        size: 4
+      - id: ls
+        type: str
+        size: 6
+  length_text_info:
+    seq:
+      - id: ltsh
+        type: str
+        size: 4
+      - id: lt
+        type: str
+        size: 5
+  length_data_info:
+    seq:
+      - id: ldsh
+        type: str
+        size: 4
+      - id: ld
+        type: str
+        size: 9
+  length_reserved_info:
+    seq:
+      - id: lrsh
+        type: str
+        size: 4
+      - id: lr
+        type: str
+        size: 7
+  image_segment:
+    params:
+      - id: idx
+        type: u2
+    instances:
+      has_mask:
+        value: image_sub_header.img_compression.substring(0, 1) == 'M' or image_sub_header.img_compression.substring(1, 2) == 'M'
+    seq:
+      - id: image_sub_header
+        type: image_sub_header
+      - id: image_data_mask
+        type: image_data_mask
+        if: has_mask
+      - id: image_data_field
+        size: _parent.header.linfo[idx].li.to_i - image_data_mask.total_size
+  image_sub_header:
+    seq:
+      - id: file_part_type
+        -orig-id: im
+        contents: 'IM'
+      - id: image_id_1
+        -orig-id: iid1
+        type: str
+        size: 10
+      - id: image_date_time
+        -orig-id: idatim
+        type: date_time
+        doc: 'UTC time of image acquisition in the format CCYYMMDDhhmmss: CC century, YY last two digits of the year, MM month, DD day, hh hour, mm minute, ss second'

```

Comment:
The format of the time string docs should be moved into the struct representing time.

Reply:
Moved